pkg/models/datasource_cache.go

```diff
@@ -177,6 +188,23 @@ func (ds *DataSource) GetHttpTransport() (*dataSourceTransport, error) {
 	return dsTransport, nil
 }
 
+func (ds *DataSource) sigV4Middleware(next http.RoundTripper) http.RoundTripper {
+	decrypted := ds.SecureJsonData.Decrypt()
```

Comment:
This will use cached decrypted values if available, otherwise decrypt them and cache them.

```suggestion
	decrypted := ds.DecryptedValues()
```